extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxBuilderDialog.js

```diff
@@ -0,0 +1,107 @@
+/* globals require */
+
+/*
+ * VisualEditor user interface WikiaInfobooxBuilderDialog class.
+ */
+
+/**
+ * Dialog for inserting portable infobox templates.
+ *
+ * @class
+ * @extends ve.ui.Dialog
+ *
+ * @constructor
+ * @param {Object} [config] Configuration options
+ */
+ve.ui.WikiaInfoboxBuilderDialog = function VeUiWikiaInfoboxBuilderDialog(config) {
+	// Parent constructor
+	ve.ui.WikiaInfoboxBuilderDialog.super.call(this, config);
+
+	$(window).on('infoboxBuilderReturnToVE', function (event, isOnPublish) {
```

Comment:
This is communication between AMD component and OOJS. This even is sent from PortableInfoboxBuilder/js/PortableInfoboxBuilderPonto.js https://github.com/Wikia/app/pull/10150/files#diff-a3a76bdb3cb9952ac27d2751302128edR37